src/Scene/AnimationPlayer.js

```diff
@@ -0,0 +1,203 @@
+    [
+        'Stop',
+        'Play',
+        'End',
+        'Pause',
+    ];
+
+/**
+ * AnimationPlayer
+ * It can play, pause or stop animation
+ */
+function AnimationPlayer(dom) {
+    this.dom = dom;
+    this.id = null;
+    this.keyframe = 0;
+    this.frameEvent = new CustomEvent('frameAnimation');
+    this.stopEvent = new CustomEvent('stopAnimation');
+    this.endEvent = new CustomEvent('endAnimation');
+    this.animation = null;
+    this.resolve = null;
+    this.promise = new Promise((r) => { this.resolve = r; });
+    this.state = PLAYER_STATE.STOP;
+}
+
+AnimationPlayer.prototype = Object.create(AnimationPlayer.prototype);
+
+AnimationPlayer.prototype.constructor = AnimationPlayer;
+
+/**
+ * Gets the interval time.
+ *
+ * @return     {Number}  The interval time is the time between consecutive frames
+ */
+AnimationPlayer.prototype.getIntervalTime = function () {
+    return INTERVAL_TIME;
+};
+
+// if debugAnimation is true console.log are enabled to sniff animation'state
+var _DEBUG = null;
+
+if (debugAnimation) {
+    _DEBUG = function (message, animation) {
+        /* eslint-disable no-alert, no-console */
+        if (animation) {
+            console.info('Animation ', message, ' : ', animation.name);
+        }
+        /* eslint-enable no-alert, no-console */
+    };
+} else {
+    _DEBUG = function () {};
+}
+
+// Private functions
+// stop timer and re-init parameter
+var resetTimer = function () {
+    if (this.id) {
+        clearInterval(this.id);
+        this.id = undefined;
+    }
+
+    this.keyframe = 0;
+};
+
+// finish animation and re-init parameter
+var finishAnimation = function () {
+    resetTimer.bind(this)();
+    this.animation = null;
+    this.resolve();
+};
+
+var setPlayerState = function (state) {
+    this.state = state;
+    _DEBUG(debugMsg[state], this.animation);
+};
+
+// Public functions
+AnimationPlayer.prototype.isPlayed = function () {
+    return this.state === PLAYER_STATE.PLAY;
+};
+
+AnimationPlayer.prototype.isStopped = function () {
+    return this.state === PLAYER_STATE.STOP;
+};
+
+AnimationPlayer.prototype.isEnded = function () {
+    return this.state === PLAYER_STATE.END;
+};
+
+/**
+ * { Start animation }
+ *
+ * @param      {Animation} The animation to play
+ * @return     {Promise}  Promise is resolved when animation is stopped or finished
+ */
+AnimationPlayer.prototype.play = function (animation) {
+    this.animation = animation;
+    setPlayerState.bind(this)(PLAYER_STATE.PLAY);
+    resetTimer.bind(this)();
+    this.id = setInterval(this.frame.bind(this), INTERVAL_TIME);
+    return new Promise((r) => { this.resolve = r; });
+};
+
+/**
+ * { Stop current animation }
+ *
+ * @return  {Promise}  Promise is resolved when animation is stopped or finished
+ */
+AnimationPlayer.prototype.stop = function () {
+    setPlayerState.bind(this)(PLAYER_STATE.STOP);
+    finishAnimation.bind(this)();
+    this.dom.dispatchEvent(this.stopEvent);
+    return this.promise;
```

Comment:
So I have a problem with `this.promise`:

- you init it in the constructor and capture the resolve function in `this.resolve`
- you return the promise only in `stop`
- you change `this.resolve` in `play` but not `this.promise`
- you resolve it in `finishAnimation`

As I expect `play` to be called before `stop` or `finishAnimation`, I don't see how `this.promise` could ever be resolved...